src/components/components/EditBenefitForm.js

```diff
@@ -0,0 +1,161 @@
+import React from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import apiBeneficio from "../../services/apiBeneficio";
+import { withRouter } from "react-router-dom";
+
+
+
+class EditBenefitForm extends React.Component {
+
+    constructor(props){
+      super(props);
+      this.state = {
+        title: "",
+        description: "",
+        price:  null,
+        redeem_way: "" ,
+        quantity: null
+      };
+
+      this.id = this.props.match.params.benefitId;
+
+      // this.history = useHistory();
+  
+      this.handleChange = this.handleChange.bind(this);
+      this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    async getBenefits(){
+      const response = await apiBeneficio.get("benefits/"+this.id);
+      this.setState(response.data);
+    }
+
+    componentDidMount(){
+      this.getBenefits();
+    }
+  
+    handleChange(event){
+      switch(event.target.id){
+        case  "title":
+          if(event.target.value.length<=20){
+            this.setState({title: event.target.value});
+          }
+          break;
+        case "description":
+          if(event.target.value.length<=20){
+            this.setState({description: event.target.value});
+          }
+          break;
+        case "price":
+          const regex = /^[0-9]|[.,]?$/gm;
+          if(event.target.value.match(regex)!=null){
+            this.setState({price: event.target.value});  
+          }else{
+            this.setState({price: this.state.price});
+          }
+          break;
+        case "redeem_way":
+          if(event.target.value.length<=25){
+            this.setState({redeem_way: event.target.value});
+          }
+          break;
+        case "quantity":
+          this.setState({quantity: event.target.value});
+          break;
+        default: break;
+      }
+    }
+  
+    async handleSubmit(event){
+      event.preventDefault();
+      const benefit = this.state;
+      
+      try{
```

Comment:
Codacy found an issue: [Identifier 'redeem_way' is not in camel case.](https://app.codacy.com/gh/fga-eps-mds/2020.1-VC_Gestor-FrontEnd/pullRequest?prid=6403093)